packages/core/strapi/lib/core/plugins/create-plugin.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { join } = require('path');
+const { isFunction, defaultsDeep } = require('lodash/fp');
+const { env } = require('@strapi/utils');
+const { createContentType } = require('@strapi/utils').contentTypes;
+const { validateStrapiServer } = require('./validation');
+
+const createPlugin = async (strapi, name, path) => {
+  const loadPluginServer = require(join(path, 'strapi-server.js'));
+  const userPluginConfig = strapi.config.get(`plugins.${name}`);
+  const pluginServer = await loadPluginServer(strapi);
+
+  await validateStrapiServer(pluginServer);
+
+  const defaultConfig = isFunction(pluginServer.config)
+    ? await pluginServer.config({ env })
```

Comment:
we could reverse the logic like so. wdyt ?

`strapi-server.js`
```js
module.exports = strapi => {

  return {
    config: strapi.loaders.loadConfig('./config'), 
  }
}
```
that would allow reusing the same logic as in the app-configuration loader and allow loading an entire config folder more easily. not sure that's necessary though

Reply:
I see, however if we are going this way we would not need to return `{ config, contentTypes }` at the end, right ?
Then it would only be a registration logic, not a declarative one. It's good for me but you were not really in favor of it if I remember correctly. wdyt ?

Reply:
actually it would only help the developer load the config but we still need to return them.